One heisen test failed in a slightly odd way:

```
self = <airflow.executors.dask_executor.DaskExecutor object at 0x7efe33cd1090>

    def end(self) -> None:
        if not self.client:
            raise AirflowException(NOT_STARTED_MESSAGE)
        if not self.futures:
>           raise AirflowException(NOT_STARTED_MESSAGE)
E           airflow.exceptions.AirflowException: The executor should be started first!

airflow/executors/dask_executor.py:113: AirflowException
```


Do we think it's safe to merge anyway?

--- a/airflow/executors/dask_executor.py
+++ b/airflow/executors/dask_executor.py
@@ -100,7 +100,7 @@ class DaskExecutor(BaseExecutor):
             self.futures.pop(future)
 
     def sync(self) -> None:
-        if not self.futures:
+        if self.futures is None:
             raise AirflowException(NOT_STARTED_MESSAGE)
         # make a copy so futures can be popped during iteration
         for future in self.futures.copy():
@@ -109,14 +109,14 @@ class DaskExecutor(BaseExecutor):
     def end(self) -> None:
         if not self.client:
             raise AirflowException(NOT_STARTED_MESSAGE)
-        if not self.futures:
+        if self.futures is None:
             raise AirflowException(NOT_STARTED_MESSAGE)
         self.client.cancel(list(self.futures.keys()))
         for future in as_completed(self.futures.copy()):
             self._process_future(future)
 
     def terminate(self):
-        if not self.futures:
+        if self.futures is None:
             raise AirflowException(NOT_STARTED_MESSAGE)
         self.client.cancel(self.futures.keys())
         self.end()

